### PR TITLE
Plane: update servo.cpp - crow brakes- diff. spoilers

### PR DESCRIPTION
**Pull request by jgilbertfpv**
added crow braking via flap channel to differential spoilers and reversed dspoiler outputs because having the inner elevon go up and outer go down causes tip stall. now inner goes down, outer goes up. still could use a percentage adjustment to change how much throw each surface gets when using flap input.

### CODE REVIEW

**Comment by jgilbertfpv** [comment]
any feedback on this?

**Comment by OXINARF** [comment]
Hi @jgilbertfpv, thank you for your first contribution! Sorry for the delay in giving feedback, our main Plane maintainer (@tridge) is quite busy at the moment but I'll try to get his attention to this as soon as possible.

**Comment by jgilbertfpv** [comment]
@OXINARF Great! Thank you!

File: ArduPlane/servos.cpp [hunk 1]
```diff
@@ -205,15 +205,24 @@ void Plane::dspoiler_update(void)
...
         // apply rudder to left wing
-        dspoiler1_left = constrain_float(elevon_left + rudder, -4500, 4500);
-        dspoiler2_left = constrain_float(elevon_left - rudder, -4500, 4500);
-    }
+        dspoiler1_left = constrain_float(elevon_left - rudder, -4500, 4500);
+        dspoiler2_left = constrain_float(elevon_left + rudder, -4500, 4500);
+    }
+	if (flap_percent > 0) {
+		//apply crow brakes to both wings
+	    dspoiler1_left = constrain_float(elevon_left + flap_percent * 25, -4500, 4500);
```
Comment (magicrub) [inline ArduPlane/servos.cpp]: why hardcoded *25 and *45?

**Comment by Kelly-Foster** [comment]
Thanks jgilbertfpv.  I look forward to trying this once it's in Plane master.

Kelly

File: ArduPlane/servos.cpp [hunk 2]
```diff
@@ -205,15 +205,24 @@ void Plane::dspoiler_update(void)
     float dspoiler2_left = elevon_left;
     float dspoiler1_right = elevon_right;
     float dspoiler2_right = elevon_right;
+    RC_Channel *flapin = RC_Channels::rc_channel(g.flapin_channel-1);
```
Comment (WickedShell) [inline ArduPlane/servos.cpp]: You need to check that a flapin channel was actually found. This could be nullptr.

**Comment by rmackay9** [comment]
Just a small thing - the commit message should start with "Plane:"

File: ArduPlane/servos.cpp (hunk 2 above)
Comment (jgilbertfpv) [reply ArduPlane/servos.cpp]: Unfortunately I'm useless after this point. I looked over another mix that used flap input and just took a couple of pieces from those lines and worked them into the differential spoilers mix and kept recompiling and flashing until everything moved the right directions.

**Comment by tridge** [comment]
rebased the PR branch

**Comment by jgilbertfpv** [comment]
I gave up on this happening- just using inner elevons as 15% of outers now.

**Comment by jgilbertfpv** [comment]
Oh wow! Can't wait to test this. Thank you!

**Comment by tridge** [comment]
I've now tested this in realflight and fixed it so that you can have both rudder and flaps with differential spoilers

**Comment by jgilbertfpv** [comment]
Great! Thank you for taking the time to mess with this. I'm happy to test in real world conditions.

**Comment by tridge** [comment]
@jgilbertfpv thanks! I'd love to hear of test results
maybe you could do a blog post on discuss.ardupilot.org? That would let others know of the new feature

**Comment by jgilbertfpv** [comment]
@tridge Sure thing, I'll try to give it a go tomorrow and see what I can come up with

**Comment by jgilbertfpv** [comment]
@tridge Are these changes here http://firmware.ardupilot.org/Plane/latest/PX4/?

**Comment by jgilbertfpv** [comment]
@tridge I have one little problem holding me back right now. everything moves correctly except for left rudder input makes diff spoiler 2 and 4 work opposite than it should. every other input moves correct. 
from the documentation this is how I have my channels set up- I had tick the checkbox to reverse the left side servos to make them move the correct way for normal flight, this also makes the flaps work correctly. right rudder input does make the right side crow the correct way. I dont think i can simply switch my two left channels because then the throws would be slightly off and flap would be backwards. suggestions?
Differential Spoiler Left 1 | 16 | This should be set to the outer left servo
Differential Spoiler Right 1 | 17 | This should be set to the outer right servo
Differential Spoiler Left 2 | 86 | This should be set to the inner left servo
Differential Spoiler Right 2 | 87 | This should be set to the outer right servo

**Comment by tridge** [comment]
@jgilbertfpv you're absolutely right, thanks for pointing this out!
I've fixed it in master, please try again and let me know

**Pull request merged**